### PR TITLE
Add Eslint v8 support

### PR DESCRIPTION
**Pull request by scsinke**
I tried upgrading to Eslint v8. This should work. I needed to modify the tests to work with the new eslint version. Let me know what you think.

This pull request should solve #17

### CODE REVIEW

**Comment by scsinke** [comment]
@marvinhagemeister

**Comment by rschristian** [comment]
What's the reason behind switching the tests to TS?

Not quite sure I see the value in dragging in TS & `ts-jest` for a 40-odd line file with only 2 type annotations.

**Comment by scsinke** [comment]
Yeah I had some trouble getting it to work with `js`. But now it's working I can for sure revert it back.

**Comment by rschristian** [comment]
Ah, gotcha. You don't need to revert it back on my opinion alone, we can see what other people think

My two cents would be that if you do keep `@types/node`, drop it down to match the CI (12.x). Don't want hints being provided for functions that may not exist when the CI runs.

**Comment by zactopus** [comment]
What can we do to get this merged? Need any help?

**Comment by scsinke** [comment]
In my opinion it should be ready. But it's awaiting approval from one of the maintainers

**Review by rschristian — APPROVED** [review]
lgtm

**Comment by Tadwork** [comment]
I was just trying to upgrade to ESLint 8 and saw this PR, I am so glad it is almost ready! 🎉

File: package.json [hunk 1]
```diff
@@ -50,18 +50,20 @@
     "@babel/plugin-syntax-class-properties": "^7.12.13",
     "@babel/plugin-syntax-decorators": "^7.12.13",
     "@babel/plugin-syntax-jsx": "^7.12.13",
-    "eslint-plugin-compat": "^3.5.1",
-    "eslint-plugin-jest": "^23.7.0",
+    "eslint-plugin-compat": "^4.0.0",
+    "eslint-plugin-jest": "^25.2.4",
     "eslint-plugin-react": "^7.0.0",
     "eslint-plugin-react-hooks": "^4.2.0"
```
Comment (Tadwork) [inline package.json:56]: should this be set to to 4.3.0 explicitly for ESLint 8 support ? https://github.com/facebook/react/blob/main/packages/eslint-plugin-react-hooks/CHANGELOG.md#430

Comment (scsinke) [reply package.json:56]: I think this should also count for `eslint-plugin-react` then? That version should be `^7.27.1`?

Comment (scsinke) [reply package.json]: @Tadwork Please check if the changes are correct.

Comment (Tadwork) [reply package.json]: @scsinke this looks great, thank you! I also tried it locally and it works great

**Review by Tadwork — APPROVED** [review]
Pulled and checked locally and works for me with eslint 8

**Review by marvinhagemeister — APPROVED** [review]
Awesome, thank you so much for the PR 🙌

**Pull request merged**